src/test/regress/sql/pg14.sql

```diff
@@ -77,7 +77,14 @@ SELECT rebalance_table_shards('col_compression', rebalance_strategy := 'by_shard
 CALL citus_cleanup_orphaned_shards();
 SELECT result AS column_compression FROM run_command_on_workers($$SELECT ARRAY(
 SELECT attname || ' ' || attcompression FROM pg_attribute WHERE attrelid::regclass::text LIKE 'pg14.col\_compression%' AND attnum > 0 ORDER BY 1
-)$$) ORDER BY length(result);
+)$$);
+
+-- test propagation of ALTER TABLE .. ALTER COLUMN .. SET COMPRESSION ..
+ALTER TABLE col_compression ALTER COLUMN b SET COMPRESSION pglz;
+ALTER TABLE col_compression ALTER COLUMN a SET COMPRESSION default;
+SELECT result AS column_compression FROM run_command_on_workers($$SELECT ARRAY(
+SELECT attname || ' ' || attcompression FROM pg_attribute WHERE attrelid::regclass::text LIKE 'pg14.col\_compression%' AND attnum > 0 ORDER BY 1
+)$$);
```

Comment:
Let's add a test for `alter table heap_table add column b text compression pglz` too, I guess `AT_SetCompression` should cover that as well.

Should we also have some simple tests with `CREATE TABLE`, `CREATE TABLE LIKE` & `CREATE TABLE AS` in another pr ?

Reply:
> Should we also have some simple tests with CREATE TABLE, CREATE TABLE LIKE & CREATE TABLE AS in another pr ?

As far as I remember, distributed Citus doesn't act on these commands. Maybe are they only relevant for columnar?

Reply:
Actually I meant to say a `create_distributed_table` command run for a table, some of whose columns are compressed.
Or a `CREATE TABLE partition of` command run against a parent distributed table.
In such cases, I guess we would attempt to build table creation commands since we will distribute the new table.

Not sure if we create a distributed table for `CREATE TABLE LIKE` run against a distributed table.

Reply:
Oh, we added those tests in https://github.com/citusdata/citus/pull/5222/files

Reply:
Looks like CREATE TABLE LIKE / AS doesn't inherit column compression already, not sure if it's a pg bug or the intended behavior.

```sql
create table heap_table (a text compression pglz);
\d+ heap_table
                                       Table "public.heap_table"
┌────────┬──────┬───────────┬──────────┬─────────┬──────────┬─────────────┬──────────────┬─────────────┐
│ Column │ Type │ Collation │ Nullable │ Default │ Storage  │ Compression │ Stats target │ Description │
├────────┼──────┼───────────┼──────────┼─────────┼──────────┼─────────────┼──────────────┼─────────────┤
│ a      │ text │           │          │         │ extended │ pglz        │              │             │
└────────┴──────┴───────────┴──────────┴─────────┴──────────┴─────────────┴──────────────┴─────────────┘
Access method: heap

create table heap_table_1 (like heap_table);
\d+ heap_table_1
                                      Table "public.heap_table_1"
┌────────┬──────┬───────────┬──────────┬─────────┬──────────┬─────────────┬──────────────┬─────────────┐
│ Column │ Type │ Collation │ Nullable │ Default │ Storage  │ Compression │ Stats target │ Description │
├────────┼──────┼───────────┼──────────┼─────────┼──────────┼─────────────┼──────────────┼─────────────┤
│ a      │ text │           │          │         │ extended │             │              │             │
└────────┴──────┴───────────┴──────────┴─────────┴──────────┴─────────────┴──────────────┴─────────────┘
Access method: heap

create table heap_table_2 as table heap_table;
\d+ heap_table_2
                                      Table "public.heap_table_2"
┌────────┬──────┬───────────┬──────────┬─────────┬──────────┬─────────────┬──────────────┬─────────────┐
│ Column │ Type │ Collation │ Nullable │ Default │ Storage  │ Compression │ Stats target │ Description │
├────────┼──────┼───────────┼──────────┼─────────┼──────────┼─────────────┼──────────────┼─────────────┤
│ a      │ text │           │          │         │ extended │             │              │             │
└────────┴──────┴───────────┴──────────┴─────────┴──────────┴─────────────┴──────────────┴─────────────┘
Access method: heap
```

Reply:
> Or a CREATE TABLE partition of command run against a parent distributed table.
In such cases, I guess we would attempt to build table creation commands since we will distribute the new table.

This might be interesting to add @halilozanakgul

Reply:
(just realized that `CREATE TABLE AS` doesn't give an option to inherit things like compression, so we can skip it)

Reply:
Added some more tests.
> Let's add a test for alter table heap_table add column b text compression pglz too, I guess AT_SetCompression should cover that as well.

Just for correctness, I believe this is actually AT_AddColumn.

Reply:
Looks good to me, anything blocking the merge of this PR? I think we should be good to merge.